server/s3_object_versioning_helper.cc

```diff
@@ -54,7 +54,7 @@ std::string S3ObjectVersioingHelper::get_versionid_from_epoch_time(
 std::string S3ObjectVersioingHelper::generate_keyid_from_versionid(
     std::string versionid) {
   // add padding if requried;
-  versionid = versionid.append((3 - versionid.size() % 3) % 3, '=');
+  versionid = versionid.append((4 - (versionid.size() & 3)) & 3, '=');
```

Comment:
@evgeniy-brazhnikov can you explain a little on why this change ?

Reply:
x % 4 === x & 3 for each integer x.
So formula is (4 - (versionid.size() % 4)) %4
There was the error before.
Bitwise AND is much faster then division.

Reply:
ok.